File: OrangeJuice.Server.Core/Configuration/GoogleAuthOptionsFactory.cs
namespace OrangeJuice.Server.Configuration
{
	public sealed class GoogleAuthOptionsFactory : Factory.IFactory<GoogleAuthOptions>
	{
		private readonly IConfigurationProvider _configurationProvider;

		public GoogleAuthOptionsFactory(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public GoogleAuthOptions Create()
		{
			return new GoogleAuthOptions
			{
				Audience = _configurationProvider.GetValue("google:Audience"),
				CertificateKey = _configurationProvider.GetValue("google:CertificateKey"),
				CertificateSecret = _configurationProvider.GetValue("google:CertificateSecret"),
				ClientId = _configurationProvider.GetValue("google:ClientId"),
				Issuer = _configurationProvider.GetValue("google:Issuer")
			};
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/GoogleAuthOptionsFactory.fs
namespace OrangeJuice.Server.FSharp.Configuration

open OrangeJuice.Server.Configuration

type GoogleAuthOptionsFactory(configurationProvider : IConfigurationProvider) =
    interface Factory.IFactory<GoogleAuthOptions>
        with member this.Create() : GoogleAuthOptions =
            new GoogleAuthOptions(
                Audience = configurationProvider.GetValue("google:Audience"),
                CertificateKey = configurationProvider.GetValue("google:CertificateKey"),
                CertificateSecret = configurationProvider.GetValue("google:CertificateSecret"),
                ClientId = configurationProvider.GetValue("google:ClientId"),
                Issuer = configurationProvider.GetValue("google:Issuer"))
